include/oneapi/dpl/internal/binary_search_impl.h

```diff
@@ -247,8 +263,10 @@ oneapi::dpl::__internal::__enable_if_execution_policy<Policy, OutputIterator>
 binary_search(Policy&& policy, InputIterator1 start, InputIterator1 end, InputIterator2 value_start,
```

Comment:
@rarutyun Can we move all `__enable_if_execution_policy` checks from interface level into special implementation of `__select_backend()` function (for the case when the `Policy` type isn't known for use `oneapi` execution policy?

Reply:
If I understand correctly, the answer is "No, we cannot" because according to the standard the overload with `ExecutionPolicy` should not participate in overload resolution if the first param is not `ExecutionPolicy`.  That's why we need the enable_if that creates SFINAE case if condition is not `true`.

Reply:
OK